src/libmongoc/src/mongoc/mongoc-opts.c

```diff
@@ -23,6 +23,8 @@ _mongoc_insert_one_opts_parse (
 {
    bson_iter_t iter;
 
+   BSON_ASSERT_PARAM (client);
+
```

Comment:
mongoc-opts.c is generated by `build/generate-opts.py`. Suggest updating [mongoc-opts.c.template](https://github.com/mongodb/mongo-c-driver/blob/572fd74d70b2294deaff7e760d2c0175099cd331/build/opts_templates/mongoc-opts.c.template#L20) and regenerating mongoc-opts.c with `python ./build/generate-opts.py`.

Reply:
What should I be updating in the template? I searched for `ASSERT`s but didn't find any.

Reply:
Add `BSON_ASSERT_PARAM (client);` to the template that generates the function:

```
bool
_{{ struct_name }}_parse (
   mongoc_client_t *client,
   const bson_t *opts,
   {{ struct_type }} *{{ struct_name }},
   bson_error_t *error)
{
```

And regenerate mongoc-opts.c by running `python ./build/generate-opts.py`.